docs/Account sign up (Account Protection) - Approved and rejected sign up flows.md

```diff
@@ -0,0 +1,126 @@
+- In the sample site, an existing user's username is set to their email address which is also tied to how they log in to the sample site. Furthermore, the sample site ensures that no two customers have the same email address. This ensures usernames will be unique for the sample site, as well as unique when sent to Dynamics 365 Fraud Protection. Decide what format to use for existing user IDs. It does not have to be an email address, but you should be careful not to duplicate user IDs.
+
+## Required data
+- Sign up ID
+- Customer local date
+- Merchant local date
+- Session ID
+- Username
+- Password hash
+
+## Optional data
+- User data (user ID, email, address, payment instrument, etc.)
+- SSO Authentication Provider
+
+## Sign up event
+This example request sends a sign up event to Dynamics 365 Fraud Protection asking for a risk assessment of the sign up.
+```http
+POST <Merchant API Endpoint>/v0.5/merchantservices/AccountProtection/events/<Merchant Instance ID>/AccountCreation/<Sign Up ID>
+Host: <Merchant API Endpoint>
+Authorization: bearer <token>
+Content-Type: application/json; charset=utf-8
+Content-Length: <content length>
+x-ms-correlation-id: <correlation ID 1>
+
+{
+  "name": "AP.AccountCreation",
+  "version": "0.5",
+  "metadata": {
+    "name": "AP.AccountCreation.Metadata",
+    "signUpId": "34f47dc4-9781-4033-99fd-185649c4b001",
+    "customerLocalDate": "2020-02-10T21:53:27.8760689-08:00",
+    "trackingId": "e8a2f3cf-d3ef-4631-a276-893665c6cf54",
+    "merchantTimeStamp": "2020-02-10T21:53:27.8822492-08:00"
+  },
+  "device": {
+    "sessionId": "b2d36c49-e2ea-422d-acff-04798b85d520",
+    "ipAddress": null,
+    "provider": null,
+    "externalDeviceId": null,
+    "externalDeviceType": null
+  },
+  "user": {
+    "userId": null,
+    "userType": "consumer",
+    "username": "user_name",
+    "passwordHash": "3q4w5e6r",
+    "firstName": "Don",
+    "lastName": "Joe",
+    "country": "us",
+    "zipCode": "98052",
+    "timeZone": "PST",
+    "language": "en-us",
+    "membershipId": null,
+    "isMembershipIdUsername": false
+  },
+  "ssoAuthenticationProvider": {
+    "authenticationProvider": "MSA | Facebook | PSN | MerchantAuth | Google",
+    "displayName": "customer display name"
+  },
+  "email": {
+    "emailType": "primary",
+    "emailValue": "abc@my.com",
+    "isEmailValidated": false,
+    "emailValidatedDate": "0001-01-01T00:00:00",
+    "isEmailUsername": false
+  },
+  "phone": {
+    "phoneType": "alternative",
+    "phoneNumber": "123-456-7890",
+    "isPhoneNumberValidated": false,
+    "phoneNumberValidatedDate": "2020-02-10T21:53:27.8833043-08:00",
+    "isPhoneUsername": false
+  },
+  "address": {
+    "addressType": "primary",
+    "firstName": null,
+    "lastName": null,
+    "phoneNumber": null,
+    "street1": null,
```

Comment:
any reason some of these address fields aren't filled in with examples even though sample site sends them?  So, as a general rule when creating these sample payloads I would run the sample app, capture the requests in Fiddler and then tweak them.  That tweaking involved me taking some liberties like adding in extra fields I thought might be useful to explain... but I didn't generally add too much more than what sample site already sends for those events.

Reply:
I got the sample payloads for the AP events from the Word doc.  looks like that doc is not the best resource, so I've based the sample payloads off the payload that the sample site sends.